go/processor/floria/run_context.go

```diff
@@ -76,9 +76,11 @@ func (r runContext) Call(kind tosca.CallKind, parameters tosca.CallParameters) (
 	}
 
 	snapshot := r.CreateSnapshot()
-	if err := transferValue(r, parameters.Value, parameters.Sender, recipient); err != nil {
-		r.RestoreSnapshot(snapshot)
-		return tosca.CallResult{}, nil
+	if kind != tosca.DelegateCall && kind != tosca.CallCode {
```

Comment:
From how I understand the semantic of `CALLCODE` it should also perform the value transfer just like CALL would do. The difference is that in the `CALLCODE` case the sender and receiver are the same. If a local transfer from a sender to itself can never fail, skipping the transfer here is fine. But if it needs to fail if there are  insufficient funds, you can not skip it here.

Reply:
@simonlechner  why is this case (callcode with insufficient funds)  not detected in the unit test. the unit test is called ?

Reply:
@LuisPH3 there were no tests with value transfers for all call types as they are implemented in the same function. With the newly added changes to the run context different call types are now handled differently and this PR adds tests to ensure this.

Reply:
This is not yet answering my question: In case of a `CallCode`, do we need to call `transferValue` to make sure there is enough balance on the account for the call?

Right now, I guess our interpreter implementations check this pro-actively (e.g. [here](https://github.com/Fantom-foundation/Tosca/blob/ce72f47093f896f9ff5528315060a2cb2df80a17/go/interpreter/lfvm/instructions.go#L1170-L1180)). But I do not think we should rely on this being implemented by all interpreters consistently.

All your tests pass if the condition is restricted to
```
if kind != tosca.DelegateCall {
```

Reply:
I reworked the integration test dealing with insufficient balance, it now also checks whether the code has been executed. The transaction should fail if a `CallCode` is executed with a value greater than the balance of the sender.